### ISSUE TITLE
Automatically added docker ports cause deployment error

### ISSUE THREAD

**Issue by ScottGuymer** [cloudfoundry-community/terraform-provider-cloudfoundry#287]
When updating an app resource that deploys a docker container I get an error around ports.

`The app is invalid: Ports must be in the 1024-65535 range.`

This seems be because the cf platform is automatically picking up ports detailed in `EXPOSE` commands in the docker container and adding them to the `ports` on the app.

When deploying the first time I only specify port `8080` in the terraform resource. But when the app is running i can see it has `[ 80, 8080]` in the ports.

When doing a subsequent deploy the provider throws an error that there is a port below the 1024 value.

I think this is relatively new behaviour in the cf platform.

Should this check be removed? Is this only for docker containers?

**Comment by ArthurHlt** [comment]
Actually, when you use a docker image with an expose, cloud foundry see it and use it, after app creation you get the port(s) from expose and the provider store it for consistency purpose. If you ran an update with your port 8080 it send to api the 80 port too because it was set by the api from expose.

Here, you should not set 8080, as it will not use 8080 for the route but the port from expose.
In the provider part, we should maybe just not send ports under 1024 to the api

**Comment by ScottGuymer** [comment]
I think in the past it did not have the behaviour to return this port in the API to the provider when refreshed. 

So now that the provider is seeing it is trying to send it back and its own validation is causing a failure.

My understanding is that cf only uses the last `EXPOSE` command that it finds in the image. In this case this is `8080` but `80` is defined before that.

**Comment by ArthurHlt** [comment]
Maybe you didn't have any `EXPOSE` before, i did that on this PR: https://github.com/cloudfoundry-community/terraform-provider-cloudfoundry/pull/261 
If it really due to cloud foundry not giving you before, maybe it is because your cloud foundry has been updated ?

**Comment by ScottGuymer** [comment]
Yeah, I think that it has been updated and this is new behaviour that I have not seen before on our instance.

Looking at the PR you mentioned reminds me that the reason I used `8080` was for the health check. There were originally `EXPOSE 80` in the image (im using standard nginx) I added `EXPOSE 8080` to fix the above. This hasn't changed in a long time.

It also worth noting that I'm using blue/green deployment.
